test/jest/system/snyk-test/python/snyk-test-pyproject.spec.ts

```diff
@@ -0,0 +1,318 @@
+          platform: undefined,
+          path: fixturePath,
+        };
+        expect(result).toMatchObject({
+          result: JSON.stringify(expectedResultObject, null, 2),
+        });
+
+        done();
+      });
+    });
+  });
+
+  describe('--all-projects flag is used to scan the project', () => {
+    describe('project does not contain poetry.lock file', () => {
+      it('should not attempt to scan poetry vulnerabilities', async (done) => {
+        const fixturePath = join(
+          __dirname,
+          'fixtures',
+          'pyproject-without-poetry',
+        );
+        const plugin = {
+          async inspect() {
+            return {
+              plugin: {
+                targetFile: 'Pipfile',
+                name: 'snyk-python-plugin',
+                runtime: 'Python',
+              },
+              package: {},
+            };
+          },
+        };
+        mockedLoadPlugin.mockImplementationOnce(() => {
+          return plugin;
+        });
+        mockedMakeRequest.mockImplementationOnce(() => {
+          return {
+            res: { statusCode: 200 },
+            body: {
+              result: { issuesData: {}, affectedPkgs: {} },
+              meta: { org: 'test-org', isPublic: false },
+              filesystemPolicy: false,
+            },
+          };
+        });
+
+        const result: CommandResult = await test(fixturePath, {
+          allProjects: true,
+          json: true,
+        });
+
+        expect(mockedLoadPlugin).toHaveBeenCalledTimes(1);
+        expect(mockedLoadPlugin).toHaveBeenCalledWith('pip');
+
+        expect(mockedMakeRequest).toHaveBeenCalledTimes(1);
+        expect(mockedMakeRequest).toHaveBeenCalledWith(
+          expect.objectContaining({
+            body: expect.objectContaining({
+              displayTargetFile: 'Pipfile',
+            }),
+          }),
+        );
+
+        const expectedResultObject = {
+          vulnerabilities: [],
+          ok: true,
+          dependencyCount: 0,
+          org: 'test-org',
+          policy: undefined,
+          isPrivate: true,
+          licensesPolicy: null,
+          packageManager: 'pip',
+          projectId: undefined,
+          ignoreSettings: null,
+          docker: undefined,
+          summary: 'No known vulnerabilities',
+          severityThreshold: undefined,
+          remediation: undefined,
+          filesystemPolicy: false,
+          uniqueCount: 0,
+          targetFile: 'Pipfile',
+          projectName: undefined,
+          foundProjectCount: undefined,
+          displayTargetFile: 'Pipfile',
+          platform: undefined,
+          path: fixturePath,
+        };
+        expect(result).toMatchObject({
+          result: JSON.stringify(expectedResultObject, null, 2),
+        });
+
+        done();
+      });
+    });
+
+    describe('project does contain poetry.lock file', () => {
+      it('should scan poetry vulnerabilities', async (done) => {
```

Comment:
```suggestion
      it('should scan python vulnerabilities', async (done) => {
```
If there's no lockfie, it should not detect a poetry project